kaznet/apps/users/tests/test_models.py

```diff
@@ -20,30 +26,279 @@ def test_userprofile_model_creation(self):
-        mommy.make('main.Submission', _quantity=789, user=userprofile.user)
+        mommy.make("main.Submission", _quantity=789, user=userprofile.user)
 
         # user .objects so that we can get the submission count
         profile = UserProfile.objects.get(id=userprofile.id)
 
         self.assertEqual(789, profile.submission_count)
+
+    def test_approved_submissions(self):
+        """
+        Test that the correct count of approved submissions is returned
+        """
+        userprofile = mommy.make("auth.User").userprofile
+
+        mommy.make(
+            "main.Submission",
+            _quantity=200,
+            user=userprofile.user,
+            status=Submission.APPROVED,
+        )
+        mommy.make("main.Submission", _quantity=300, user=userprofile.user)
+
+        self.assertEqual(userprofile.approved_submissions, 200)
+
+    def test_rejected_submissions(self):
+        """
+        Test that the correct count of rejected submissions is returned
+        """
+        userprofile = mommy.make("auth.User").userprofile
+
+        mommy.make(
+            "main.Submission",
+            _quantity=200,
+            user=userprofile.user,
+            status=Submission.REJECTED,
+        )
+        mommy.make("main.Submission", _quantity=300, user=userprofile.user)
+
+        self.assertEqual(userprofile.rejected_submissions, 200)
+
+    def test_approval_rate(self):
+        """
+        Test that the correct approval rate is returned
+        """
+        userprofile = mommy.make("auth.User").userprofile
+
+        # when there are no submissions, the approval rate is 0
+        self.assertEqual(userprofile.approval_rate, 0.0)
+
+        approved_count = 200
+        default_count = 300
+        mommy.make(
+            "main.Submission",
+            _quantity=approved_count,
+            user=userprofile.user,
+            status=Submission.APPROVED,
+        )
+        mommy.make("main.Submission", _quantity=default_count,
+                   user=userprofile.user)
+        approval_rate = approved_count / (approved_count + default_count)
+
+        # when there are submission
+        self.assertEqual(userprofile.approval_rate, approval_rate)
+
+    def test_amount_earned(self):
+        """
+        Test that the correct ammount earned is returned
+        """
+        userprofile = mommy.make("auth.User").userprofile
+        bounty_ammount_sum = 0
+        for _ in range(20):
+            bounty = mommy.make("main.Bounty",
+                                amount=random.randrange(100, 1000))
+            bounty_ammount_sum += bounty.amount
+            mommy.make(
+                "main.Submission",
+                user=userprofile.user,
+                bounty=bounty,
+                status=Submission.APPROVED,
+            )
+        # check if the submission gives the correct amount_earned
+        self.assertEqual(userprofile.amount_earned.amount, bounty_ammount_sum)
+
+    @staticmethod
```

Comment:
For Django tests, the convention for this kind of thing is to make it available as a property/method on the `self` object.  You set that up in the `setUp` method of the test class.  Therefore, please remove this staticmethod.

See: https://docs.djangoproject.com/en/2.2/topics/testing/tools/